From 640e873cf26fe0f1bd0a479a02c1e252520b6379 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 8 Oct 2024 14:15:23 +0200
Subject: [PATCH 1277/1695] plat-stm32mp1: allow probe on non-secure timers for
 stm32mp15 platforms

On stm32mp15x platforms, the CSI/HSI clocks calibration is based on timers
that cannot be secured but only MCU isolated. Therefore, allow the probe
of these devices for these platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 392cd8b3c95a6783e68b942f7cbe940adcb52857 plat-stm32mp1: allow probe on non-secure timers for stm32mp15 platforms]
Change-Id: Ifdec99d35593c10acc32e7c992909f1b5a94d709
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/409757
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/main.c b/core/arch/arm/plat-stm32mp1/main.c
index 4ae590756..9268b7f9c 100644
--- a/core/arch/arm/plat-stm32mp1/main.c
+++ b/core/arch/arm/plat-stm32mp1/main.c
@@ -850,7 +850,8 @@ bool stm32mp_allow_probe_shared_device(const void *fdt, int node)
 	 * with non-secure world
 	 */
 	if (node == uart_console_node ||
-	    !strcmp(compat, "st,stm32mp15-i2c-non-secure"))
+	    !strcmp(compat, "st,stm32mp15-i2c-non-secure") ||
+	    (IS_ENABLED(CFG_STM32MP15) && !strcmp(compat, "st,stm32-timers")))
 		return true;
 
 	return false;
-- 
2.47.2

